File: guet/steps/check/check.py
from typing import List

from guet.steps.step import Step


class Check(Step):
    def __init__(self, stop_message: str = None):
        super().__init__()
        self._stop_message = stop_message

    def do_play(self, args: List[str]):
        if self.should_stop(args):
            message = self._stop_message
            if message is None:
                message = self.load_message(args)
            print(message)
            exit(1)

    def should_stop(self, args: List[str]) -> bool:
        raise NotImplementedError()

    def load_message(self, _: List[str]) -> str:
        return ""

File: src/steps/check/check.ts
import { Step } from '../step';
import { Args } from '../../args';
import { print } from 'gluegun';

export abstract class Check extends Step {
    private stopMessage?: string;
    constructor(stopMessage: string) {
        super();
        this.stopMessage = stopMessage;
    }

    public doPlay(args: Args): void {
        if (this.shouldStop(args)) {
            this.printErrorMessage(args);
            process.exit(1);
        }
    }

    private printErrorMessage(args: Args) {
        this.stopMessage
            ? print.info(this.stopMessage)
            : print.info(this.stopMessageHook(args));
    }

    public abstract shouldStop(args: Args): boolean;

    public stopMessageHook(args: Args): string {
        return '';
    }
}
